certificate-service/domain/src/main/java/se/inera/intyg/certificateservice/domain/validation/model/ElementValidationDiagnosis.java

```diff
@@ -0,0 +1,129 @@
+package se.inera.intyg.certificateservice.domain.validation.model;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.Value;
+import se.inera.intyg.certificateservice.domain.certificate.model.ElementData;
+import se.inera.intyg.certificateservice.domain.certificate.model.ElementValue;
+import se.inera.intyg.certificateservice.domain.certificate.model.ElementValueDiagnosis;
+import se.inera.intyg.certificateservice.domain.certificate.model.ElementValueDiagnosisList;
+import se.inera.intyg.certificateservice.domain.certificatemodel.model.ElementId;
+import se.inera.intyg.certificateservice.domain.certificatemodel.model.FieldId;
+import se.inera.intyg.certificateservice.domain.diagnosiscode.model.DiagnosisCode;
+import se.inera.intyg.certificateservice.domain.diagnosiscode.repository.DiagnosisCodeRepository;
+
+@Value
+@Getter(AccessLevel.NONE)
+@Builder
+public class ElementValidationDiagnosis implements ElementValidation {
+
+  private static final String DIAGNOSIS_DESCRIPTION = ".description";
+  FieldId mandatoryField;
+  List<FieldId> order;
+  DiagnosisCodeRepository diagnosisCodeRepository;
+
+  @Override
+  public List<ValidationError> validate(ElementData data,
+      Optional<ElementId> categoryId) {
+    if (data == null) {
+      throw new IllegalArgumentException("Element data is null");
+    }
+
+    final var elementValueDiagnosisList = getValue(data.value());
+
+    if (mandatoryField != null && elementValueDiagnosisList.diagnoses().isEmpty()) {
+      return List.of(
+          errorMessage(data, mandatoryField, categoryId, "Ange minst en diagnos.")
+      );
+    }
+
+    if (mandatoryField != null && missingMandatoryField(elementValueDiagnosisList)) {
+      return List.of(
+          errorMessage(data, mandatoryField, categoryId, "Ange diagnos på översta raden först.")
```

Comment:
This is not a general message since mandatory field could be set to something else than first row

Reply:
True, the message is not general.

Diagnosis always has main diagnosis first tho. If this behaviour change I think we should update how the error message looks. But is it fine for now to not use a general message?

Reply:
After discussion, we added a todo and will handle it later.